[WIP] Enhance: 3rd party resource handling

sprocketc opened the pull request:
Continuing on #9761

- [x] **reveal.js** - View page as slides to test this
- [x] **shepherd.js** - Clear all app data and create a new graph to test a popover onboarding
- [x] **mhchem**

logseq-cldwalker commented:
@sprocketc Looks like a dnd.spec test has failed twice on this branch which doesn't fail on master. Happy to review this when tests are passing

sprocketc commented:
@logseq-cldwalker I had to change the behaviour of drag n drop to fix this, although just increasing the `0.75` factor of the failing test would work. I added my understanding of how blocks' dnd should work to the test file
> When we drag and drop a block, it should always be moved under the target element, unless the targer is the first element of its container. In that case, if we drop it at the top half of the target, it should be moved on top of it. 

The changes on the code and the tests reflect that. Also, `getOffsetRect` seems to be unneeded, since the properties returned by [`getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) are relative to the viewport. I think @tiensonqin and @andelf should confirm all of the above before we merge this. The commit in question https://github.com/logseq/logseq/pull/9907/commits/36335106b491bef64fc631b3d145ad36e22938cc

sprocketc commented:
I found the underlying issue. The previously bundled reveal css applied some css rules to the `html` element
![Screenshot from 2023-07-24 13-14-57](https://github.com/logseq/logseq/assets/10744960/2117f756-0d3b-4cbd-94f1-9ab779ee8467)
The current one is not. Installing `v4.0.2` did not fix this. It looks like there was a mismatch between js and css versions. Setting the html overflow to hidden fixed the tests https://github.com/logseq/logseq/pull/9907/commits/a5d21377487b1e5ec636dc36b9bc621aa37cfa05 The problem was that on small screen sizes (<= 1024px wide), there is a slight x and y overflow, enabling the horizontal and vertical scrollbars. The e2e tests run at a 1024x768px virtual display. That affected some tests that required precise mouse movement.

@xyhp915 I also removed the html height rule here https://github.com/logseq/logseq/pull/9907/commits/3bce8310273a82b6fcb3be1257a943a550847bcf, because there is no height rule to unset anymore.

I reverted the dnd changes, mentioned on the previous comment. Although those changes made the behavior more robust and independent of page scrolling, they don't belong in this PR. I'll open a new one to address this specific issue.

logseq-cldwalker reviewed (approved):
@sprocketc QAed all 3 changes and they work great! 👍 🚢 Thanks for the conversion and digging into the underlying issue with reveal
